azure/services/agentpools/spec.go

```diff
@@ -191,6 +334,7 @@ func (s *AgentPoolSpec) Parameters(ctx context.Context, existing interface{}) (p
 				NodeTaints:          existingPool.NodeTaints,
 				Tags:                existingPool.Tags,
 				KubeletConfig:       existingPool.KubeletConfig,
+				LinuxOSConfig:       existingPool.LinuxOSConfig,
```

Comment:
If these settings are all immutable as enforced by the webhook, we shouldn't need to define this on `existingProfile` or `normalizedProfile` since those are only initialized with values that can change.

Reply:
Yes, thanks for pointing it out!